release/models/acl/openconfig-packet-match.yang

```diff
@@ -462,12 +470,97 @@ module openconfig-packet-match {
           to match the destination port";
     }
 
-    leaf-list tcp-flags {
+    leaf detail-mode {
+      type enumeration {
+        enum EXPLICIT {
+          description
+            "Specifies that the mode for matching details at the transport
+             layer is to explicitly match transport flags.";
+        }
+        enum BUILTIN {
+          description
+            "Specifies that the mode for matching details at the transport
+            layer is to using implementation built-ins which may map to
+            multiple flags.";
+        }
+      }
+      description
+        "Mode that is used for matching detailed fields at the transport
+        layer. When EXPLICIT is specified, the implementation should
+        match based on the explicit flags that are specified in the
+        corresponding leaf. When BUILTIN is specified, the implementation
+        must expand the contents of the corresponding leaf to the flags
+        and/or fields that match the pre-defined built-in values.";
+    }
+
+    leaf explicit-detail-match-mode {
+      type enumeration {
+        enum ANY {
+          description
+            "Matches of the explicit-detail-flags field are treated as
+            an OR between the values in the list.";
+        }
+        enum ALL {
+          description
+            "Matches of the explicit-details-flags field are treated
+            as an AND of the values in the list.";
+        }
+      }
+      description
+        "Specifies how the contents of the explicit-details-flags list
+        are to be treated. ANY implies that any of the flags may match,
+        where ALL indicates that all the flags must be matched.";
+      when "../detail-mode = 'EXPLICIT'" {
+        description
+          "This leaf is only valid when the mode for matches is specified to
+          be explicit.";
+      }
+    }
+
+    leaf-list explicit-tcp-flags {
       type identityref {
         base oc-pkt-match-types:TCP_FLAGS;
       }
       description
-        "List of TCP flags to match";
+        "An explicit list of the TCP flags that are to be matched. The
+        mechanism for the match is specified by the explicit-detail-match-mode
+        leaf.";
+      when "../detail-mode = 'EXPLICIT'" {
+        description
+          "This leaf is only valid when the mode for matches is specified to
+          be explicit.";
+      }
+    }
+
+    leaf builtin-detail {
+      type enumeration {
+        enum TCP_INITIAL {
+          description
+            "Matches the first packet of a TCP session based on a packet
+            not having the ACK flag set, and having the SYN flag set.";
+        }
+        enum TCP_ESTABLISHED {
+          description
+            "Matches an established TCP session based on a packet having
+            the ACK or RST flags set. This does not match the first
+            packet.";
+        }
+        enum FRAGMENT {
```

Comment:
It is illogical to define IP FRAGMENTATION match as /acl/acl-sets/acl-set/acl-entries/acl-entry/transport/state/builtin-detail

-- "FRAGMENT" - Matches non-zero values of the fragment-offset field" of IPv4 HEADER. Not L4/TCP/UDP/ICMP/....
- FRAGMENT is not applicable for IPv6 (in contrast to any other transport leaf)
It shall be under /acl/acl-sets/acl-set/acl-entries/acl-entry/ipv4/ perhaps "FRAGMENTATION" list of booleans: with 2 Leafs: "FIRST_FRAGMENT" (more fragments bit set && offser == 0), "FOLLOWING_FRAGMENT" (offset > 0). logic among this 2 should be OR.